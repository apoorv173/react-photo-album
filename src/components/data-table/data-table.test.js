import React from "react";
import DataTable from "./data-table";
import { shallow, mount } from "enzyme";

describe("Data table  comnponent", () => {

    const onCountChange = jest.fn();

    it("Should render the data table", () => {
        const tree = shallow(<DataTable 
                onCountChange={onCountChange}
                rowCount={20} 
            ></DataTable>)
        expect(tree).toMatchSnapshot();
    });
    

    it('should render all the components isnide the data table component', () => {
        const wrapper = mount(<DataTable 
            onCountChange={onCountChange}
            rowCount={20} 
        ></DataTable>);
        expect(wrapper.find('.d-inline-block').length).toBe(2);
    });

    it('should call the on on previous and on next event', () => {
        const wrapper = mount(<DataTable 
            onCountChange={onCountChange}
            rowCount={20} 
        ></DataTable>);

        wrapper.find(".select-wrapper").simulate("change");
        expect(onCountChange).toHaveBeenCalled();

    });
});