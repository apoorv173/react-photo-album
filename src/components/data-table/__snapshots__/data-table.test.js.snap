// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data table  comnponent Should render the data table 1`] = `ShallowWrapper {}`;

exports[`Data table  comnponent Should render the modal window with both the buttons disabled 1`] = `ShallowWrapper {}`;
